tests/integration/test_charm_agent.py

```diff
@@ -0,0 +1,175 @@
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Tests for jenkins agents."""
+
+import asyncio
+
+import jenkins
+from ops.model import Application, ActiveStatus
+import pytest
+import pytest_asyncio
+from pytest_operator.plugin import OpsTest
+
+
+async def install_jenkins_version(
+    ops_test: OpsTest, app: Application, jenkins_version: str
+) -> None:
+    """Install a specific version of Jenkins into the charm.
+
+    Args:
+        ops_test: pytest operator reference to the test environment.
+        app: The charm running Jenkins.
+        jenkins_version: The version of Jenkins to install in the charm.
+
+    """
+    # Install dependencies
+    action = await app.units[0].run_action("install-dependencies", jenkins_version=jenkins_version)
+    await action.wait()
+    await ops_test.model.wait_for_idle(status=ActiveStatus.name)
+
+    # Install specified Jenkins version
+    await app.units[0].ssh(
+        "sudo apt-get install -y --allow-downgrades jenkins={}".format(jenkins_version)
+    )
+    apt_cache_output = await app.units[0].ssh("apt-cache policy jenkins")
+    assert "Installed: {}".format(jenkins_version) in apt_cache_output
+
+    # Restart Jenkins
+    await app.units[0].ssh("sudo systemctl restart jenkins")
+
+
+@pytest_asyncio.fixture(
+    scope="module",
+    params=[
+        pytest.param(None, id="latest LTS jenkins version"),
+        # The following are the versions of Jenkins running in production in Canonical as of
+        # 2022-09-21
+        pytest.param("2.361.1", id="jenkins version 2.361"),
+        pytest.param("2.346.3", id="jenkins version 2.346"),
+        pytest.param("2.332.4", id="jenkins version 2.332"),
+        pytest.param("2.277.4", id="jenkins version 2.277"),
+        pytest.param("2.263.4", id="jenkins version 2.263"),
+        pytest.param("2.249.3", id="jenkins version 2.249"),
+        pytest.param("2.235.5", id="jenkins version 2.235"),
+        pytest.param("2.222.4", id="jenkins version 2.222"),
+        pytest.param("2.176.4", id="jenkins version 2.176"),
+        pytest.param("2.150.3", id="jenkins version 2.150"),
+    ],
+)
+async def app_jenkins_version(ops_test: OpsTest, app: Application, request: pytest.FixtureRequest):
+    """Install a range of jenkins versions to run the tests against."""
+    jenkins_version = request.param
+    if jenkins_version:
+        install_jenkins_version(ops_test=ops_test, app=app, jenkins_version=jenkins_version)
+
+    yield app
+
+
+@pytest_asyncio.fixture(scope="function")
+async def agent(ops_test: OpsTest):
+    """Deploy machine agent and destroy it after tests complete."""
+    agent: Application = await ops_test.model.deploy(
+        # Currently hardcoded due to the charm on charmhub not currently working, needs to be
+        # updated once a working version of the agent machine charm is deployed
+        "/home/jdkandersson/src/jenkins-agent-charm/"
```

Comment:
This will not work in Github Action, provide this through a command line argument?

Reply:
Thank you, fixed